mirrord-agent/src/file.rs

```diff
@@ -36,10 +47,28 @@ impl FileManager {
         );
 
         OpenOptions::from(open_options)
-            .open(path)
+            .open(&path)
             .map(|file| {
-                let fd = std::os::unix::prelude::AsRawFd::as_raw_fd(&file);
-                self.open_files.insert(fd, file);
+                // let fd = std::os::unix::prelude::AsRawFd::as_raw_fd(&file);
+                let fd = get_fd();
+
+                let _ = file
+                    .metadata()
+                    .map_err(|err| {
+                        error!(
+                            "FileManager::open -> Failed to get metadata for file {:#?}",
+                            err
+                        );
+                        err
+                    })
+                    .map(|metadata| {
+                        debug!("FileManager::open -> Got metadata for file {:#?}", metadata);
+                        if metadata.is_dir() {
```

Comment:
Stylewise, I would suggest
```suggestion
                        let value = if metadata.is_dir() { RemoteFile::Directory(path)} else {RemoteFile::File(file)};
                        self.open_files.insert(fd, value);
```